plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -3185,12 +3199,32 @@ protected String getIoUringCheckCommand() {
      * (i) Qemu >= 5.0;
      * (ii) Libvirt >= 6.3.0
      */
-    protected void setDiskIoDriver(DiskDef disk) {
-        if (enableIoUring) {
+    public void setDiskIoDriver(DiskDef disk, boolean iothreadsEnabled, IoDriver ioDriver) {
+        if (iothreadsEnabled) {
+            disk.setIothreads(iothreadsEnabled);
+            if (!enableIoUring && IoDriver.IOURING == ioDriver) {
+                //setting io=threads as default
+                disk.setIoDriver(IoDriver.THREADS);
+            } else {
+                disk.setIoDriver(ioDriver);
+            }
+        } else if (enableIoUring) {
             disk.setIoDriver(DiskDef.IoDriver.IOURING);
```

Comment:
Applies here as well. Try to make this uniform and have it work the same is iothreads now works?